tests/Microsoft.Identity.Test.Unit.net45/OAuthClientTests.cs

```diff
@@ -164,6 +167,56 @@ public void OAuthClient_FailsWithServiceExceptionWhenResponseIsHttpNotFound()
                 });
         }
 
+#if DESKTOP
+        [TestMethod]
+        public void PKeyAuthSuccsesResponseTest()
+        {
+            using (var harness = CreateTestHarness())
+            {
+                harness.HttpManager.AddInstanceDiscoveryMockHandler();
+
+                PublicClientApplication app = PublicClientApplicationBuilder.Create(TestConstants.ClientId)
+                                                                            .WithAuthority(new Uri(ClientApplicationBase.DefaultAuthority), true)
+                                                                            .WithHttpManager(harness.HttpManager)
+                                                                            .WithTelemetry(new TraceTelemetryConfig())
+                                                                            .BuildConcrete();
+
+                MsalMockHelpers.ConfigureMockWebUI(
+                    app.ServiceBundle.PlatformProxy,
+                    AuthorizationResult.FromUri(app.AppConfig.RedirectUri + "?code=some-code"));
+
+                harness.HttpManager.AddMockHandlerForTenantEndpointDiscovery(TestConstants.AuthorityCommonTenant);
+
+                //Initiates PKeyAuth challenge which will trigger an additional request sent to AAD to satisfy the PKeyAuth challenge
+                harness.HttpManager.AddMockHandler(
+                    new MockHttpMessageHandler
+                    {
+                        ExpectedMethod = HttpMethod.Post,
+                        ResponseMessage = MockHelpers.CreatePKeyAuthChallengeResponse()
+                    });
+                harness.HttpManager.AddSuccessTokenResponseMockHandlerForPost(TestConstants.AuthorityCommonTenant);
+
+                AuthenticationResult result = app
```

Comment:
This test only checks that an additional request to the token endpoint is made. It does not check what headers are sent, any of the NetDesktopDeviceManager functionality etc.